syntax = "proto3";

package sqlInjection;

enum Sex {
  Man = 0;
  Female = 1;
}

message User {
  string name = 1;
  string family = 2;
  uint64 id = 3;
  uint64 age = 4;
  Sex sex = 5;
  string createdAt = 6;
}

message GetUsersRequest {
  optional uint64 userId = 1;
  string authKey = 2;
  uint64 messageId = 3;
}

message GetUsersWithSqlInjectRequest {
  optional string userId = 1;
  string authKey = 2;
  uint64 messageId = 3;
}

message GetUsersResponse {
  repeated User users = 1;
  uint64 messageId = 2;
}

service SQLService {

  rpc getUsers(GetUsersRequest) returns (GetUsersResponse);

  rpc getUsersWithSqlInject(GetUsersWithSqlInjectRequest) returns (GetUsersResponse);
}
